tests/unit/test_utilities.py

```diff
@@ -358,3 +361,53 @@ def test_make_array_two_dimensional(self):
 
         self.assertEqual(len(output_array.shape), 2)
         np.testing.assert_array_equal(output_array, expected_output)
+
+
+class TestTransposeIfXdimLessThanYdim(TestCase):
+
+    def test_wider_than_tall(self):
+        # Test case where x dim <  y dim and should transpose
+        input_array = np.ma.array([[1, 2, 3], [4, 5, 6]])
+        expected_output = np.ma.array([[1, 4], [2, 5], [3, 6]])
+        result = transpose_if_xdim_less_than_ydim(input_array)
+        np.testing.assert_array_equal(result, expected_output)
+        self.assertEqual(result.shape, (3, 2))
+
+    def test_taller_than_wide(self):
+        # Test case where x < y and should not transpose.
+        input_array = np.ma.array([[1, 2], [3, 4], [5, 6]])
+        result = transpose_if_xdim_less_than_ydim(input_array)
+        np.testing.assert_array_equal(result, input_array)
+        self.assertEqual(result.shape, (3, 2))
+
+    def test_square_array(self):
+        # test where y dim == x dim and should not transpose.
+        input_array = np.ma.array([[1, 2], [3, 4]])
+        result = transpose_if_xdim_less_than_ydim(input_array)
+        np.testing.assert_array_equal(result, input_array)
+        self.assertEqual(result.shape, (2, 2))
+
+    def test_1d_array(self):
+        # Test case with a 1D array
+        input_array = np.ma.array([1, 2, 3])
+        with self.assertRaisesRegex(ValueError, 'variable must be 2 dimensional'):
+            transpose_if_xdim_less_than_ydim(input_array)
+
+    def test_3d_array(self):
+        # Test case with a 3D array
+        input_array = np.ma.array([[[1, 2], [3, 4]], [[5, 6], [7, 8]]])
+        with self.assertRaisesRegex(ValueError, 'variable must be 2 dimensional'):
```

Comment:
I appreciate that these tests are comprehensive enough to exercise not just the happy-path, but also the ways in which things might fail.